OpenRA.Game/Map/Map.cs

```diff
@@ -853,6 +853,9 @@ public WDist DistanceAboveTerrain(WPos pos)
 			var cell = CellContaining(pos);
 			var offset = pos - CenterOfCell(cell);
 
+			if (!Contains(cell))
```

Comment:
This should be `Ramp.Contains(cell)`. `Map.Contains` behaves differently, and is a lot slower.

Reply:
Fixed.